guacamole-ext/src/main/java/org/apache/guacamole/properties/GuacamoleProperty.java

```diff
@@ -49,5 +51,25 @@
      *                            provided value.
      */
     public Type parseValue(String value) throws GuacamoleException;
+    
+    /**
+     * Parses the given string value into a sorted Collection of values of the
+     * type associated with this GuacamoelProperty.
+     * 
+     * @param value
+     *     The string value to parse.
+     * 
+     * @return
+     *     A sorted Collection of the parsed values.
+     * 
+     * @throws GuacamoleException 
+     *     If an error occurs while parsing the provided value.
+     */
+    default public Collection<Type> parseValueCollection(String value)
+            throws GuacamoleException {
+        
+        throw new GuacamoleServerException("Collection parsing not implemented for this property type.");
```

Comment:
This should be a `GuacamoleUnsupportedException`. Another option could be something like `Collections.singletonList(parseValue(value))`.